src/Microsoft.ML.LightGBM.StaticPipe/LightGbmStaticExtensions.cs

```diff
@@ -44,17 +45,53 @@ public static class LightGbmStaticExtensions
             int? numLeaves = null,
             int? minDataPerLeaf = null,
             double? learningRate = null,
-            int numBoostRound = LightGbmArguments.Defaults.NumBoostRound,
-            Action<LightGbmArguments> advancedSettings = null,
+            int numBoostRound = Options.Defaults.NumBoostRound,
             Action<LightGbmRegressionModelParameters> onFit = null)
         {
-            CheckUserValues(label, features, weights, numLeaves, minDataPerLeaf, learningRate, numBoostRound, advancedSettings, onFit);
+            CheckUserValues(label, features, weights, numLeaves, minDataPerLeaf, learningRate, numBoostRound, onFit);
 
             var rec = new TrainerEstimatorReconciler.Regression(
                (env, labelName, featuresName, weightsName) =>
                {
                    var trainer = new LightGbmRegressorTrainer(env, labelName, featuresName, weightsName, numLeaves,
-                       minDataPerLeaf, learningRate, numBoostRound, advancedSettings);
+                       minDataPerLeaf, learningRate, numBoostRound);
+                   if (onFit != null)
+                       return trainer.WithOnFitDelegate(trans => onFit(trans.Model));
+                   return trainer;
+               }, label, features, weights);
+
+            return rec.Score;
+        }
+
+        /// <summary>
+        /// Predict a target using a tree regression model trained with the <see cref="LightGbmRegressorTrainer"/>.
+        /// </summary>
+        /// <param name="catalog">The <see cref="RegressionCatalog"/>.</param>
+        /// <param name="label">The label column.</param>
+        /// <param name="features">The features column.</param>
+        /// <param name="weights">The weights column.</param>
+        /// <param name="options">Algorithm advanced settings.</param>
+        /// <param name="onFit">A delegate that is called every time the
+        /// <see cref="Estimator{TInShape, TOutShape, TTransformer}.Fit(DataView{TInShape})"/> method is called on the
+        /// <see cref="Estimator{TInShape, TOutShape, TTransformer}"/> instance created out of this. This delegate will receive
+        /// the linear model that was trained. Note that this action cannot change the result in any way;
+        /// it is only a way for the caller to be informed about what was learnt.</param>
+        /// <returns>The Score output column indicating the predicted value.</returns>
+        public static Scalar<float> LightGbm(this RegressionCatalog.RegressionTrainers catalog,
+            Scalar<float> label, Vector<float> features, Scalar<float> weights,
```

Comment:
`FastTree` just has `catalog` and `options` for its input parameters when there is an explicit `options` field. Is there a standard for what the "`options` API" looks like? If not, let's make one. This needs to be consistent across the learners.

Reply:
These are the `static` extensions.   

---
In reply to: [251533791](https://github.com/dotnet/machinelearning/pull/2261#discussion_r251533791) [](ancestors = 251533791)